src/main/java/com/gdmc/httpinterfacemod/handlers/HeightmapHandler.java

```diff
@@ -0,0 +1,125 @@
+                JsonArray jsonRow = new JsonArray();
+                for (int val : row) {
+                    jsonRow.add(val);
+                }
+                responseArray.add(jsonRow);
+            }
+
+            // Respond with that array as a string
+            Headers responseHeaders = httpExchange.getResponseHeaders();
+            setDefaultResponseHeaders(responseHeaders);
+            resolveRequest(httpExchange, responseArray.toString());
+        }
+        else {
+            throw new HttpException("Method not allowed. Only GET or POST requests are supported.", 405);
+        }
+    }
+
+
+    public int[][] getHeightmap(BuildAreaHandler.BuildArea buildArea, ServerLevel serverlevel, Types heightmapType) {
+
+        // Get the x/z size of the build area
+        var xSize = buildArea.getxTo() - buildArea.getxFrom() + 1;
+        var zSize = buildArea.getzTo() - buildArea.getzFrom() + 1;
+        // Create the 2D array to store the heightmap data
+        var heightmap = new int[xSize][zSize];
+
+        // Get the number of chunks
+        int xChunkCount = Math.floorDiv(buildArea.getxTo(), 16) - Math.floorDiv(buildArea.getxFrom(), 16) + 1;
+        int zChunkCount = Math.floorDiv(buildArea.getzTo(), 16) - Math.floorDiv(buildArea.getzFrom(), 16) + 1;
+
+        // Get the chunk x and z of the chunk at the lowest x and z
+        int minChunkX = Math.floorDiv(buildArea.getxFrom(), 16);
+        int minChunkZ = Math.floorDiv(buildArea.getzFrom(), 16);
+
+        // For every chunk in the build area
+        for (int chunkX = minChunkX; chunkX < xChunkCount + minChunkX; ++chunkX) {
+            for (int chunkZ = minChunkZ; chunkZ < zChunkCount + minChunkZ; ++chunkZ) {
+
+                // Get the chunk
+                var chunk = serverlevel.getChunk(chunkX, chunkZ);
+
+                // Get the heightmap of the appropriate type
+                HashMap<Types, Heightmap> heightmaps = new HashMap<>();
+                chunk.getHeightmaps().forEach(item -> heightmaps.put(item.getKey(), item.getValue()));
+                var chunkHeightmap = heightmaps.get(heightmapType);
+
+                // For every combination of x and z in that chunk
+                int chunkMinX = chunkX * 16;
+                int chunkMinZ = chunkZ * 16;
+                for (int x = chunkMinX; x < chunkMinX + 16; ++x) {
+                    for (int z = chunkMinZ; z < chunkMinZ + 16; ++z) {
+                        // If the column is out of bounds skip it
+                        if (x < buildArea.getxFrom() || x > buildArea.getxTo() || z < buildArea.getzFrom() || z > buildArea.getzTo()) {
+                            continue;
+                        }
+                        // Set the value in the heightmap array
+                        heightmap[x - buildArea.getxFrom()][z - buildArea.getzFrom()] = chunkHeightmap.getHighestTaken(x - chunkMinX, z - chunkMinZ);
+                    }
+                }
+            }
+        }
```

Comment:
I see 2… 4… 5 loops here. I'm pretty sure this could rewritten to be faster with less loops.

You may be able to save on converting between absolute positions and section/chunk positions by using the `ChunkPos` and `SectionPos` classes.